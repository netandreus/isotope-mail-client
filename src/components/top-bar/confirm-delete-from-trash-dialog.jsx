import React from 'react';
import PropTypes from 'prop-types';
import {translate} from 'react-i18next';
import Dialog from '../dialog/dialog';

export const ConfirmDeleteFromTrashDialog = ({t, visible, cancelAction, deleteAction}) => {
  const actions = [
    {label: t('topBar.deleteFromTrashDialog.cancel'), action: cancelAction},
    {label: t('topBar.deleteFromTrashDialog.delete'), action: deleteAction}
  ];
  return (
    <Dialog
      visible={visible}
      title={t('topBar.deleteFromTrashDialog.title')}
      actions={actions}
    >
      <p>{t('topBar.deleteFromTrashDialog.message')}</p>
    </Dialog>
  );
};

ConfirmDeleteFromTrashDialog.propTypes = {
  visible: PropTypes.bool.isRequired,
  cancelAction: PropTypes.func.isRequired,
  deleteAction: PropTypes.func.isRequired
};

export default (translate()(ConfirmDeleteFromTrashDialog));
