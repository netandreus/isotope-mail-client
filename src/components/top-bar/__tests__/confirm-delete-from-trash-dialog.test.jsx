import React from 'react';
import {shallow} from 'enzyme/build/index';
import {ConfirmDeleteFromTrashDialog} from '../confirm-delete-from-trash-dialog';


describe('ConfirmDeleteFromTrashDialog component test suite', () => {
  test('Snapshot render, should render ConfirmDeleteFromTrashDialog', () => {
    // Given
    const props = {
      t: translation => (translation),
      visible: true,
      cancelAction: () => {},
      deleteAction: () => {}
    };

    // When
    const confirmDeleteFromTrashDialog = shallow(<ConfirmDeleteFromTrashDialog {...props} />);

    // Then
    expect(confirmDeleteFromTrashDialog).toMatchSnapshot();
  });
});
