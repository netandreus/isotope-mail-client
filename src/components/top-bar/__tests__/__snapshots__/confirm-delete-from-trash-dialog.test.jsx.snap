// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDeleteFromTrashDialog component test suite Snapshot render, should render ConfirmDeleteFromTrashDialog 1`] = `
<Dialog
  actions={
    Array [
      Object {
        "action": [Function],
        "label": "topBar.deleteFromTrashDialog.cancel",
      },
      Object {
        "action": [Function],
        "label": "topBar.deleteFromTrashDialog.delete",
      },
    ]
  }
  className=""
  containerClassName=""
  contentClassName=""
  scrimClick={[Function]}
  title="topBar.deleteFromTrashDialog.title"
  visible={true}
>
  <p>
    topBar.deleteFromTrashDialog.message
  </p>
</Dialog>
`;
